Source/Core/Core/IOS/FS/FileSystemProxy.h

```diff
@@ -24,12 +24,50 @@ constexpr FS::Fd INVALID_FD = 0xffffffff;
 class FSDevice : public Device
 {
 public:
+  class ScopedFd
+  {
+  public:
+    ScopedFd(FSDevice* fs, s64 fd, Ticks tick_tracker = {})
+        : m_fs{fs}, m_fd{fd}, m_tick_tracker{tick_tracker}
+    {
+    }
+
+    ~ScopedFd()
+    {
+      if (m_fd >= 0)
+        m_fs->Close(m_fd, m_tick_tracker);
+    }
+
+    ScopedFd(ScopedFd&& other) noexcept { *this = std::move(other); }
+    ScopedFd& operator=(ScopedFd&& other) noexcept
+    {
+      m_fs = other.m_fs;
+      m_fd = other.Release();
+      m_tick_tracker = other.m_tick_tracker;
+      return *this;
+    }
```

Comment:
The purpose of implementing the move constructor is to make moves do the right thing (clear the FD in the move source to prevent a double close) and make this a proper move only type, not to optimise.

Reply:
What I meant to say is that I don't believe the functions are being used now; I tried deleting them and it compiled fine. Is that on purpose and you implemented them for future support?

For the move assignment operator, suppose somebody does the following:
```
auto foobar = fs->Open(/* file foo.txt */);
foobar = fs->Open(/* file bar.txt */);
```
As implemented the descriptor for foo.txt will be left open, which might be an unintended leak of the sort ScopedFd is intended to prevent, but changing it to close foo.txt might be unexpected behavior in other cases.

I think the semantics of the assignment operator in that regard are fundamentally unclear and so it should be deleted.  It's not used at the moment, but if you want to keep the ability to assign (as opposed to copy constructing) ScopedFd values I would create CloseAndOverwrite and ReleaseAndOverwrite (or similar) and use those instead.

Reply:
Ah, good point. I'll just delete it then since it's not used at the moment.